### PR TITLE
gh-100583: Improving the `pydoc` doc

### PR DESCRIPTION
**Pull request by ramvikrams**
gh-100583: changed the doc for calling pydoc my making it platform agnostic

### CODE REVIEW

**Comment by Fidget-Spinner** [comment]
Thanks. You need to change all occurences of this command. Also we usually dont write `python you are using` like that (sorry if I misled you). Just writing `python -m pydoc` is sufficient.

**Comment by ramvikrams** [comment]
> Thanks. You need to change all occurences of this command. Also we usually dont write `python you are using` like that (sorry if I misled you). Just writing `python -m pydoc` is sufficient.

Thanks I'll do it

**Comment by slateny** [comment]
I wonder if being so explicit would instead be a bit too noisy, especially with it being bolded. [Here's](https://deploy-preview-100590--python-cpython-preview.netlify.app/library/pydoc.html) the preview for reference.

Maybe the first instance of the command can be (for example) the full `python -m pydoc sys`, and then it's implicit for subsequent mentions in the page? If too subtle, then a small note noting that `pydoc` is just an alias for `python -m pydoc` could work too.

**Comment by ramvikrams** [comment]
> 

or I was thinking to mention once `<python> -m pydoc` when running in the start and for subsequent reference we could substitute it like `pydoc_means=<python> -m pydoc`. What do you think about it.

File: Doc/library/pydoc.rst [hunk 1]
```diff
@@ -33,13 +33,13 @@ as text on the console.  The same text documentation can also be viewed from
 outside the Python interpreter by running :program:`pydoc` as a script at the
 operating system's command prompt. For example, running ::
 
-   <python you are using> -m pydoc sys
+   <python> -m pydoc sys
```
Comment (Fidget-Spinner) [inline Doc/library/pydoc.rst]: Please remove the angled brackets. So just `python -m pydoc sys`.

Same for all the other occurences below.

Comment (ramvikrams) [reply Doc/library/pydoc.rst]: yes

File: Doc/library/pydoc.rst [hunk 2]
```diff
@@ -33,13 +33,13 @@ as text on the console.  The same text documentation can also be viewed from
...
 
 at a shell prompt will display documentation on the :mod:`sys` module, in a
 style similar to the manual pages shown by the Unix :program:`man` command.  The
-argument to :program:`pydoc` can be the name of a function, module, or package,
+argument to :program:`<python> -m pydoc` can be the name of a function, module, or package,
 or a dotted reference to a class, method, or function within a module or module
-in a package.  If the argument to :program:`pydoc` looks like a path (that is,
+in a package.  If the argument to :program:`<python> -m pydoc` looks like a path (that is,
 it contains the path separator for your operating system, such as a slash in
 Unix), and refers to an existing Python source file, then documentation is
```
Comment (Fidget-Spinner) [inline Doc/library/pydoc.rst]: I don't think these need changing. Please revert the changes.

File: Doc/library/pydoc.rst [hunk 3]
```diff
@@ -51,9 +51,9 @@ produced for that file.
    executed on that occasion.  Use an ``if __name__ == '__main__':`` guard to
    only execute code when a file is invoked as a script and not just imported.
 
-When printing output to the console, :program:`pydoc` attempts to paginate the
+When printing output to the console, :program:`<python> -m pydoc` attempts to paginate the
 output for easier reading.  If the :envvar:`PAGER` environment variable is set,
-:program:`pydoc` will use its value as a pagination program.
+:program:`<python> -m pydoc` will use its value as a pagination program.
```
Comment (Fidget-Spinner) [inline Doc/library/pydoc.rst]: Please revert these changes too. These are not command examples.

File: Doc/library/pydoc.rst [hunk 4]
```diff
@@ -65,25 +65,25 @@ manner similar to the Unix :program:`man` command.  The synopsis line of a
...
+:program:`<python> -m pydoc -b` will start the server and additionally open a web
 browser to a module index page.  Each served page has a navigation bar at the
 top where you can *Get* help on an individual item, *Search* all modules with a
 keyword in their synopsis line, and go to the *Module index*, *Topics* and
 *Keywords* pages.
 
-When :program:`pydoc` generates documentation, it uses the current environment
-and path to locate modules.  Thus, invoking :program:`pydoc spam`
+When :program:`<python> -m pydoc` generates documentation, it uses the current environment
+and path to locate modules.  Thus, invoking :program:`<python> -m pydoc spam`
```
Comment (Fidget-Spinner) [inline Doc/library/pydoc.rst]: These don't need changing. Please revert the changes.

File: Doc/library/pydoc.rst [hunk 5]
```diff
@@ -65,25 +65,25 @@ manner similar to the Unix :program:`man` command.  The synopsis line of a
...
 
-:program:`pydoc -n <hostname>` will start the server listening at the given
+:program:`<python> -m pydoc -n <hostname>` will start the server listening at the given
 hostname.  By default the hostname is 'localhost' but if you want the server to
 be reached from other machines, you may want to change the host name that the
 server responds to.  During development this is especially useful if you want
 to run pydoc from within a container.
 
-:program:`pydoc -b` will start the server and additionally open a web
+:program:`<python> -m pydoc -b` will start the server and additionally open a web
```
Comment (Fidget-Spinner) [inline Doc/library/pydoc.rst]: I think we can keep these.

Comment (ramvikrams) [reply Doc/library/pydoc.rst]: updated the doc

File: Doc/library/pydoc.rst [hunk 6]
```diff
@@ -65,18 +65,18 @@ manner similar to the Unix :program:`man` command.  The synopsis line of a
 module is the first line of its documentation string.
 
 You can also use :program:`pydoc` to start an HTTP server on the local machine
-that will serve documentation to visiting web browsers.  :program:`pydoc -p 1234`
+that will serve documentation to visiting web browsers. :program:`python -m pydoc -p 1234`
```
Comment (Fidget-Spinner) [inline Doc/library/pydoc.rst]: Please leave the leading space alone. reStructuredText allows for one or two spaces after a period.

File: Doc/library/pydoc.rst (hunk 5 above)
Comment (Fidget-Spinner) [reply Doc/library/pydoc.rst]: Great! You can click "resolve conversation" after you feel that the review comment has been addressed.

File: Doc/library/pydoc.rst (hunk 6 above)
Comment (ramvikrams) [reply Doc/library/pydoc.rst]: Yes will do

**Review by Fidget-Spinner — APPROVED** [review]
This LGTM. Does anyone else have any feedback?

**Comment by ambv** [comment]
We never used `<python> -m ...` in the docs before. We always just said `python -m ...`. I think we should stick to that for consistency with the rest of the documentation.

**Review by kumaraditya303 — APPROVED** [review]

**Comment by kumaraditya303** [comment]
Thanks!

**Comment by CAM-Gerlach** [comment]
Too late now for this PR, but just FYI, the `:program:` role is only intended to be used for [the name of an [external] executable program](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-program). not the full invocation, and will render in bold, non-monospaced text. In future versions of Sphinx, with the appropriate setup, we may be able to use `:option:` for this, but for now the standard literal role, ``` `` ```, is the best option here (no pun intended).

Also, since the 3.10 branch is equivalently active to the 3.11 branch in bugfix mode, and the problem reported against both branches, I'm not sure why it wasn't marked for backport against both—for docs changes, backporting things incompletely makes it more likely to hit merge conflicts on future backports, so AFAIK we generally try to backport to all bugfix branches to which the change is applicable.
